File: mRemoteV1/UI/UI.Window.About.vb
Imports System.ComponentModel
Imports System.IO
Imports mRemote3G.App
Imports mRemote3G.Messages
Imports mRemote3G.My.Resources
Imports WeifenLuo.WinFormsUI.Docking

Namespace UI

    Namespace Window
        Public Class About
            Inherits Base

#Region "Form Init"

            Friend WithEvents lblCopyright As Label
            Friend WithEvents lblTitle As Label
            Friend WithEvents lblVersion As Label
            Friend WithEvents lblLicense As Label
            Friend WithEvents txtChangeLog As TextBox
            Friend WithEvents lblChangeLog As Label
            Friend WithEvents pnlBottom As Panel
            Friend WithEvents pbLogo As PictureBox
            Friend WithEvents lblEdition As Label
            Friend WithEvents llblFAMFAMFAM As LinkLabel
            Friend WithEvents llblMagicLibrary As LinkLabel
            Friend WithEvents llblWeifenLuo As LinkLabel
            Friend WithEvents pnlTop As Panel

            Private Sub InitializeComponent()
                Me.pnlTop = New Panel()
                Me.lblEdition = New Label()
                Me.pbLogo = New PictureBox()
                Me.pnlBottom = New Panel()
                Me.llblWeifenLuo = New LinkLabel()
                Me.llblMagicLibrary = New LinkLabel()
                Me.llblFAMFAMFAM = New LinkLabel()
                Me.txtChangeLog = New TextBox()
                Me.lblTitle = New Label()
                Me.lblVersion = New Label()
                Me.lblChangeLog = New Label()
                Me.lblLicense = New Label()
                Me.lblCopyright = New Label()
                Me.pnlTop.SuspendLayout()
                CType(Me.pbLogo, ISupportInitialize).BeginInit()
                Me.pnlBottom.SuspendLayout()
                Me.SuspendLayout()
                '
                'pnlTop
                '
                Me.pnlTop.Anchor = CType(((AnchorStyles.Top Or AnchorStyles.Left) _
                                          Or AnchorStyles.Right),
                                         AnchorStyles)
                Me.pnlTop.BackColor = Color.Black
                Me.pnlTop.Controls.Add(Me.lblEdition)
                Me.pnlTop.Controls.Add(Me.pbLogo)
                Me.pnlTop.ForeColor = Color.White
                Me.pnlTop.Location = New Point(- 1, - 1)
                Me.pnlTop.Name = "pnlTop"
                Me.pnlTop.Size = New Size(788, 145)
                Me.pnlTop.TabIndex = 0
                '
                'lblEdition
                '
                Me.lblEdition.Anchor = CType((AnchorStyles.Top Or AnchorStyles.Right), AnchorStyles)
                Me.lblEdition.BackColor = Color.Black
                Me.lblEdition.Font = New Font("Microsoft Sans Serif", 14.25!, FontStyle.Bold, GraphicsUnit.Point,
                                              CType(0, Byte))
                Me.lblEdition.ForeColor = Color.White
                Me.lblEdition.Location = New Point(509, 107)
                Me.lblEdition.Name = "lblEdition"
                Me.lblEdition.Size = New Size(264, 29)
                Me.lblEdition.TabIndex = 0
                Me.lblEdition.Text = "Edition"
                Me.lblEdition.TextAlign = ContentAlignment.BottomRight
                Me.lblEdition.Visible = False
                '
                'pbLogo
                '
                Me.pbLogo.Image = Logo
                Me.pbLogo.Location = New Point(8, 8)
                Me.pbLogo.Name = "pbLogo"
                Me.pbLogo.Size = New Size(492, 128)
                Me.pbLogo.SizeMode = PictureBoxSizeMode.AutoSize
                Me.pbLogo.TabIndex = 1
                Me.pbLogo.TabStop = False
                '
                'pnlBottom
                '
                Me.pnlBottom.Anchor = CType((((AnchorStyles.Top Or AnchorStyles.Bottom) _
                                              Or AnchorStyles.Left) _
                                             Or AnchorStyles.Right),
                                            AnchorStyles)
                Me.pnlBottom.BackColor = SystemColors.Control
                Me.pnlBottom.Controls.Add(Me.llblWeifenLuo)
                Me.pnlBottom.Controls.Add(Me.llblMagicLibrary)
                Me.pnlBottom.Controls.Add(Me.llblFAMFAMFAM)
                Me.pnlBottom.Controls.Add(Me.txtChangeLog)
                Me.pnlBottom.Controls.Add(Me.lblTitle)
                Me.pnlBottom.Controls.Add(Me.lblVersion)
                Me.pnlBottom.Controls.Add(Me.lblChangeLog)
                Me.pnlBottom.Controls.Add(Me.lblLicense)
                Me.pnlBottom.Controls.Add(Me.lblCopyright)
                Me.pnlBottom.ForeColor = SystemColors.ControlText
                Me.pnlBottom.Location = New Point(- 1, 144)
                Me.pnlBottom.Name = "pnlBottom"
                Me.pnlBottom.Size = New Size(788, 418)
                Me.pnlBottom.TabIndex = 1
                '
                'llblWeifenLuo
                '
                Me.llblWeifenLuo.AutoSize = True
                Me.llblWeifenLuo.Font = New Font("Microsoft Sans Serif", 11.0!, FontStyle.Regular, GraphicsUnit.Point,
                                                 CType(0, Byte))
                Me.llblWeifenLuo.ForeColor = SystemColors.ControlText
                Me.llblWeifenLuo.LinkColor = Color.Blue
                Me.llblWeifenLuo.Location = New Point(8, 184)
                Me.llblWeifenLuo.Name = "llblWeifenLuo"
                Me.llblWeifenLuo.Size = New Size(97, 27)
                Me.llblWeifenLuo.TabIndex = 9
                Me.llblWeifenLuo.TabStop = True
                Me.llblWeifenLuo.Text = "WeifenLuo"
                Me.llblWeifenLuo.UseCompatibleTextRendering = True
                '
                'llblMagicLibrary
                '
                Me.llblMagicLibrary.AutoSize = True
                Me.llblMagicLibrary.Font = New Font("Microsoft Sans Serif", 11.0!, FontStyle.Regular, GraphicsUnit.Point,
                                                    CType(0, Byte))
                Me.llblMagicLibrary.ForeColor = SystemColors.ControlText
                Me.llblMagicLibrary.LinkColor = Color.Blue
                Me.llblMagicLibrary.Location = New Point(8, 157)
                Me.llblMagicLibrary.Name = "llblMagicLibrary"
                Me.llblMagicLibrary.Size = New Size(115, 27)
                Me.llblMagicLibrary.TabIndex = 8
                Me.llblMagicLibrary.TabStop = True
                Me.llblMagicLibrary.Text = "MagicLibrary"
                Me.llblMagicLibrary.UseCompatibleTextRendering = True
                '
                'llblFAMFAMFAM
                '
                Me.llblFAMFAMFAM.AutoSize = True
                Me.llblFAMFAMFAM.Font = New Font("Microsoft Sans Serif", 11.0!, FontStyle.Regular, GraphicsUnit.Point,
                                                 CType(0, Byte))
                Me.llblFAMFAMFAM.ForeColor = SystemColors.ControlText
                Me.llblFAMFAMFAM.LinkColor = Color.Blue
                Me.llblFAMFAMFAM.Location = New Point(8, 130)
                Me.llblFAMFAMFAM.Name = "llblFAMFAMFAM"
                Me.llblFAMFAMFAM.Size = New Size(126, 27)
                Me.llblFAMFAMFAM.TabIndex = 4
                Me.llblFAMFAMFAM.TabStop = True
                Me.llblFAMFAMFAM.Text = "FAMFAMFAM"
                Me.llblFAMFAMFAM.UseCompatibleTextRendering = True
                '
                'txtChangeLog
                '
                Me.txtChangeLog.Anchor = CType((((AnchorStyles.Top Or AnchorStyles.Bottom) _
                                                 Or AnchorStyles.Left) _
                                                Or AnchorStyles.Right),
                                               AnchorStyles)
                Me.txtChangeLog.BackColor = SystemColors.Control
                Me.txtChangeLog.BorderStyle = BorderStyle.None
                Me.txtChangeLog.Cursor = Cursors.Default
                Me.txtChangeLog.Font = New Font("Microsoft Sans Serif", 9.0!, FontStyle.Regular, GraphicsUnit.Point,
                                                CType(0, Byte))
                Me.txtChangeLog.ForeColor = SystemColors.ControlText
                Me.txtChangeLog.Location = New Point(24, 241)
                Me.txtChangeLog.Multiline = True
                Me.txtChangeLog.Name = "txtChangeLog"
                Me.txtChangeLog.ReadOnly = True
                Me.txtChangeLog.ScrollBars = ScrollBars.Vertical
                Me.txtChangeLog.Size = New Size(760, 175)
                Me.txtChangeLog.TabIndex = 7
                Me.txtChangeLog.TabStop = False
                '
                'lblTitle
                '
                Me.lblTitle.AutoSize = True
                Me.lblTitle.Font = New Font("Microsoft Sans Serif", 14.0!, FontStyle.Bold, GraphicsUnit.Point,
                                            CType(0, Byte))
                Me.lblTitle.ForeColor = SystemColors.ControlText
                Me.lblTitle.Location = New Point(8, 16)
                Me.lblTitle.Name = "lblTitle"
                Me.lblTitle.Size = New Size(149, 33)
                Me.lblTitle.TabIndex = 0
                Me.lblTitle.Text = "mRemote3G"
                Me.lblTitle.UseCompatibleTextRendering = True
                '
                'lblVersion
                '
                Me.lblVersion.AutoSize = True
                Me.lblVersion.Font = New Font("Microsoft Sans Serif", 11.0!, FontStyle.Regular, GraphicsUnit.Point,
                                              CType(0, Byte))
                Me.lblVersion.ForeColor = SystemColors.ControlText
                Me.lblVersion.Location = New Point(8, 49)
                Me.lblVersion.Name = "lblVersion"
                Me.lblVersion.Size = New Size(71, 27)
                Me.lblVersion.TabIndex = 1
                Me.lblVersion.Text = "Version"
                Me.lblVersion.UseCompatibleTextRendering = True
                '
                'lblChangeLog
                '
                Me.lblChangeLog.AutoSize = True
                Me.lblChangeLog.Font = New Font("Microsoft Sans Serif", 11.0!, FontStyle.Regular, GraphicsUnit.Point,
                                                CType(0, Byte))
                Me.lblChangeLog.ForeColor = SystemColors.ControlText
                Me.lblChangeLog.Location = New Point(8, 211)
                Me.lblChangeLog.Name = "lblChangeLog"
                Me.lblChangeLog.Size = New Size(115, 27)
                Me.lblChangeLog.TabIndex = 6
                Me.lblChangeLog.Text = "Change Log:"
                Me.lblChangeLog.UseCompatibleTextRendering = True
                '
                'lblLicense
                '
                Me.lblLicense.AutoSize = True
                Me.lblLicense.Font = New Font("Microsoft Sans Serif", 11.0!, FontStyle.Regular, GraphicsUnit.Point,
                                              CType(0, Byte))
                Me.lblLicense.ForeColor = SystemColors.ControlText
                Me.lblLicense.Location = New Point(8, 103)
                Me.lblLicense.Name = "lblLicense"
                Me.lblLicense.Size = New Size(72, 27)
                Me.lblLicense.TabIndex = 5
                Me.lblLicense.Text = "License"
                Me.lblLicense.UseCompatibleTextRendering = True
                '
                'lblCopyright
                '
                Me.lblCopyright.AutoSize = True
                Me.lblCopyright.Font = New Font("Microsoft Sans Serif", 11.0!, FontStyle.Regular, GraphicsUnit.Point,
                                                CType(0, Byte))
                Me.lblCopyright.ForeColor = SystemColors.ControlText
                Me.lblCopyright.Location = New Point(8, 76)
                Me.lblCopyright.Name = "lblCopyright"
                Me.lblCopyright.Size = New Size(88, 27)
                Me.lblCopyright.TabIndex = 2
                Me.lblCopyright.Text = "Copyright"
                Me.lblCopyright.UseCompatibleTextRendering = True
                '
                'About
                '
                Me.BackColor = SystemColors.Control
                Me.ClientSize = New Size(784, 564)
                Me.Controls.Add(Me.pnlTop)
                Me.Controls.Add(Me.pnlBottom)
                Me.Font = New Font("Microsoft Sans Serif", 8.25!, FontStyle.Regular, GraphicsUnit.Point, CType(0, Byte))
                Me.ForeColor = SystemColors.ControlText
                Me.Icon = mRemote_Icon
                Me.MaximumSize = New Size(20000, 10000)
                Me.Name = "About"
                Me.TabText = "About"
                Me.Text = "About"
                Me.pnlTop.ResumeLayout(False)
                Me.pnlTop.PerformLayout()
                CType(Me.pbLogo, ISupportInitialize).EndInit()
                Me.pnlBottom.ResumeLayout(False)
                Me.pnlBottom.PerformLayout()
                Me.ResumeLayout(False)
            End Sub

#End Region

#Region "Public Methods"

            Public Sub New(Panel As DockContent)
                Me.WindowType = Type.About
                Me.DockPnl = Panel
                Me.InitializeComponent()
                Runtime.FontOverride(Me)
            End Sub

#End Region

#Region "Private Methods"

            Private Sub ApplyLanguage()
                lblLicense.Text = Language.Language.strLabelReleasedUnderGPL
                lblChangeLog.Text = Language.Language.strLabelChangeLog
                TabText = Language.Language.strAbout
                Text = Language.Language.strAbout
            End Sub

            Private Sub ApplyEditions()
#If PORTABLE Then
                lblEdition.Text = My.Language.strLabelPortableEdition
                lblEdition.Visible = True
#End If
            End Sub

            Private Sub FillLinkLabel(llbl As LinkLabel, Text As String, URL As String)
                llbl.Links.Clear()

                Dim Open As Integer = Text.IndexOf("[")
                Dim Close As Integer
                While Open <> - 1
                    Text = Text.Remove(Open, 1)
                    Close = Text.IndexOf("]", Open)
                    If Close = - 1 Then Exit While
                    Text = Text.Remove(Close, 1)
                    llbl.Links.Add(Open, Close - Open, URL)
                    Open = Text.IndexOf("[", Open)
                End While

                llbl.Text = Text
            End Sub

#End Region

#Region "Form Stuff"

            Private Sub About_Load(sender As Object, e As EventArgs) Handles Me.Load
                ApplyLanguage()
                ApplyEditions()

                Try
                    lblCopyright.Text = My.Application.Info.Copyright

                    Me.lblVersion.Text = "Version " & My.Application.Info.Version.ToString

                    FillLinkLabel(llblFAMFAMFAM, Language.Language.strFAMFAMFAMAttribution,
                                  Language.Language.strFAMFAMFAMAttributionURL)
                    FillLinkLabel(llblMagicLibrary, Language.Language.strMagicLibraryAttribution,
                                  Language.Language.strMagicLibraryAttributionURL)
                    FillLinkLabel(llblWeifenLuo, Language.Language.strWeifenLuoAttribution,
                                  Language.Language.strWeifenLuoAttributionURL)

                    If File.Exists(My.Application.Info.DirectoryPath & "\CHANGELOG.TXT") Then
                        Dim sR As New StreamReader(My.Application.Info.DirectoryPath & "\CHANGELOG.TXT")
                        Me.txtChangeLog.Text = sR.ReadToEnd
                        sR.Close()
                    End If
                Catch ex As Exception
                    Runtime.MessageCollector.AddMessage(MessageClass.ErrorMsg,
                                                        "Loading About failed" & vbNewLine & ex.ToString(), True)
                End Try
            End Sub

            Private Sub llblFAMFAMFAM_LinkClicked(sender As Object, e As LinkLabelLinkClickedEventArgs) _
                Handles llblFAMFAMFAM.LinkClicked
                Runtime.GoToURL(Language.Language.strFAMFAMFAMAttributionURL)
            End Sub

            Private Sub llblMagicLibrary_LinkClicked(sender As Object, e As LinkLabelLinkClickedEventArgs) _
                Handles llblMagicLibrary.LinkClicked
                Runtime.GoToURL(Language.Language.strMagicLibraryAttributionURL)
            End Sub

            Private Sub llblWeifenLuo_LinkClicked(sender As Object, e As LinkLabelLinkClickedEventArgs) _
                Handles llblWeifenLuo.LinkClicked
                Runtime.GoToURL(Language.Language.strWeifenLuoAttributionURL)
            End Sub

#End Region
        End Class
    End Namespace

End Namespace

File: mRemoteV1/UI/Windows/UI.Window.About.cs
using System;
using Microsoft.VisualBasic;
using System.Windows.Forms;
using WeifenLuo.WinFormsUI.Docking;
using System.IO;
using mRemoteNG.App;


namespace mRemoteNG.UI.Window
{
	public class About : Base
	{
        #region Form Init
		internal System.Windows.Forms.Label lblCopyright;
		internal System.Windows.Forms.Label lblTitle;
		internal System.Windows.Forms.Label lblVersion;
		internal System.Windows.Forms.Label lblLicense;
		internal System.Windows.Forms.TextBox txtChangeLog;
		internal System.Windows.Forms.Label lblChangeLog;
		internal System.Windows.Forms.Panel pnlBottom;
		internal System.Windows.Forms.PictureBox pbLogo;
		internal System.Windows.Forms.Label lblEdition;
		internal System.Windows.Forms.LinkLabel llblFAMFAMFAM;
		internal System.Windows.Forms.LinkLabel llblMagicLibrary;
		internal System.Windows.Forms.LinkLabel llblWeifenLuo;
		internal System.Windows.Forms.Panel pnlTop;
				
		private void InitializeComponent()
		{
			this.pnlTop = new System.Windows.Forms.Panel();
			this.Load += new System.EventHandler(About_Load);
			this.lblEdition = new System.Windows.Forms.Label();
			this.pbLogo = new System.Windows.Forms.PictureBox();
			this.pnlBottom = new System.Windows.Forms.Panel();
			this.llblWeifenLuo = new System.Windows.Forms.LinkLabel();
			this.llblWeifenLuo.LinkClicked += new System.Windows.Forms.LinkLabelLinkClickedEventHandler(this.llblWeifenLuo_LinkClicked);
			this.llblMagicLibrary = new System.Windows.Forms.LinkLabel();
			this.llblMagicLibrary.LinkClicked += new System.Windows.Forms.LinkLabelLinkClickedEventHandler(this.llblMagicLibrary_LinkClicked);
			this.llblFAMFAMFAM = new System.Windows.Forms.LinkLabel();
			this.llblFAMFAMFAM.LinkClicked += new System.Windows.Forms.LinkLabelLinkClickedEventHandler(this.llblFAMFAMFAM_LinkClicked);
			this.txtChangeLog = new System.Windows.Forms.TextBox();
			this.lblTitle = new System.Windows.Forms.Label();
			this.lblVersion = new System.Windows.Forms.Label();
			this.lblChangeLog = new System.Windows.Forms.Label();
			this.lblLicense = new System.Windows.Forms.Label();
			this.lblCopyright = new System.Windows.Forms.Label();
			this.pnlTop.SuspendLayout();
			((System.ComponentModel.ISupportInitialize) this.pbLogo).BeginInit();
			this.pnlBottom.SuspendLayout();
			this.SuspendLayout();
			//
			//pnlTop
			//
			this.pnlTop.Anchor = (System.Windows.Forms.AnchorStyles) ((System.Windows.Forms.AnchorStyles.Top | System.Windows.Forms.AnchorStyles.Left) 
				| System.Windows.Forms.AnchorStyles.Right);
			this.pnlTop.BackColor = System.Drawing.Color.Black;
			this.pnlTop.Controls.Add(this.lblEdition);
			this.pnlTop.Controls.Add(this.pbLogo);
			this.pnlTop.ForeColor = System.Drawing.Color.White;
			this.pnlTop.Location = new System.Drawing.Point(-1, -1);
			this.pnlTop.Name = "pnlTop";
			this.pnlTop.Size = new System.Drawing.Size(788, 145);
			this.pnlTop.TabIndex = 0;
			//
			//lblEdition
			//
			this.lblEdition.Anchor = (System.Windows.Forms.AnchorStyles) (System.Windows.Forms.AnchorStyles.Top | System.Windows.Forms.AnchorStyles.Right);
			this.lblEdition.BackColor = System.Drawing.Color.Black;
			this.lblEdition.Font = new System.Drawing.Font("Microsoft Sans Serif", (float) (14.25F), System.Drawing.FontStyle.Bold, System.Drawing.GraphicsUnit.Point, System.Convert.ToByte(0));
			this.lblEdition.ForeColor = System.Drawing.Color.White;
			this.lblEdition.Location = new System.Drawing.Point(512, 112);
			this.lblEdition.Name = "lblEdition";
			this.lblEdition.Size = new System.Drawing.Size(264, 24);
			this.lblEdition.TabIndex = 0;
			this.lblEdition.Text = "Edition";
			this.lblEdition.TextAlign = System.Drawing.ContentAlignment.BottomRight;
			this.lblEdition.Visible = false;
			//
			//pbLogo
			//
			this.pbLogo.Image = My.Resources.Logo;
			this.pbLogo.Location = new System.Drawing.Point(8, 8);
			this.pbLogo.Name = "pbLogo";
			this.pbLogo.Size = new System.Drawing.Size(492, 128);
			this.pbLogo.SizeMode = System.Windows.Forms.PictureBoxSizeMode.AutoSize;
			this.pbLogo.TabIndex = 1;
			this.pbLogo.TabStop = false;
			//
			//pnlBottom
			//
			this.pnlBottom.Anchor = (System.Windows.Forms.AnchorStyles) (((System.Windows.Forms.AnchorStyles.Top | System.Windows.Forms.AnchorStyles.Bottom) 
				| System.Windows.Forms.AnchorStyles.Left) 
				| System.Windows.Forms.AnchorStyles.Right);
			this.pnlBottom.BackColor = System.Drawing.SystemColors.Control;
			this.pnlBottom.Controls.Add(this.llblWeifenLuo);
			this.pnlBottom.Controls.Add(this.llblMagicLibrary);
			this.pnlBottom.Controls.Add(this.llblFAMFAMFAM);
			this.pnlBottom.Controls.Add(this.txtChangeLog);
			this.pnlBottom.Controls.Add(this.lblTitle);
			this.pnlBottom.Controls.Add(this.lblVersion);
			this.pnlBottom.Controls.Add(this.lblChangeLog);
			this.pnlBottom.Controls.Add(this.lblLicense);
			this.pnlBottom.Controls.Add(this.lblCopyright);
			this.pnlBottom.ForeColor = System.Drawing.SystemColors.ControlText;
			this.pnlBottom.Location = new System.Drawing.Point(-1, 144);
			this.pnlBottom.Name = "pnlBottom";
			this.pnlBottom.Size = new System.Drawing.Size(788, 418);
			this.pnlBottom.TabIndex = 1;
			//
			//llblWeifenLuo
			//
			this.llblWeifenLuo.AutoSize = true;
			this.llblWeifenLuo.Font = new System.Drawing.Font("Microsoft Sans Serif", (float) (11.0F), System.Drawing.FontStyle.Regular, System.Drawing.GraphicsUnit.Point, System.Convert.ToByte(0));
			this.llblWeifenLuo.ForeColor = System.Drawing.SystemColors.ControlText;
			this.llblWeifenLuo.LinkColor = System.Drawing.Color.Blue;
			this.llblWeifenLuo.Location = new System.Drawing.Point(16, 158);
			this.llblWeifenLuo.Name = "llblWeifenLuo";
			this.llblWeifenLuo.Size = new System.Drawing.Size(78, 22);
			this.llblWeifenLuo.TabIndex = 9;
			this.llblWeifenLuo.TabStop = true;
			this.llblWeifenLuo.Text = "WeifenLuo";
			this.llblWeifenLuo.UseCompatibleTextRendering = true;
			//
			//llblMagicLibrary
			//
			this.llblMagicLibrary.AutoSize = true;
			this.llblMagicLibrary.Font = new System.Drawing.Font("Microsoft Sans Serif", (float) (11.0F), System.Drawing.FontStyle.Regular, System.Drawing.GraphicsUnit.Point, System.Convert.ToByte(0));
			this.llblMagicLibrary.ForeColor = System.Drawing.SystemColors.ControlText;
			this.llblMagicLibrary.LinkColor = System.Drawing.Color.Blue;
			this.llblMagicLibrary.Location = new System.Drawing.Point(16, 136);
			this.llblMagicLibrary.Name = "llblMagicLibrary";
			this.llblMagicLibrary.Size = new System.Drawing.Size(92, 22);
			this.llblMagicLibrary.TabIndex = 8;
			this.llblMagicLibrary.TabStop = true;
			this.llblMagicLibrary.Text = "MagicLibrary";
			this.llblMagicLibrary.UseCompatibleTextRendering = true;
			//
			//llblFAMFAMFAM
			//
			this.llblFAMFAMFAM.AutoSize = true;
			this.llblFAMFAMFAM.Font = new System.Drawing.Font("Microsoft Sans Serif", (float) (11.0F), System.Drawing.FontStyle.Regular, System.Drawing.GraphicsUnit.Point, System.Convert.ToByte(0));
			this.llblFAMFAMFAM.ForeColor = System.Drawing.SystemColors.ControlText;
			this.llblFAMFAMFAM.LinkColor = System.Drawing.Color.Blue;
			this.llblFAMFAMFAM.Location = new System.Drawing.Point(16, 116);
			this.llblFAMFAMFAM.Name = "llblFAMFAMFAM";
			this.llblFAMFAMFAM.Size = new System.Drawing.Size(101, 22);
			this.llblFAMFAMFAM.TabIndex = 4;
			this.llblFAMFAMFAM.TabStop = true;
			this.llblFAMFAMFAM.Text = "FAMFAMFAM";
			this.llblFAMFAMFAM.UseCompatibleTextRendering = true;
			//
			//txtChangeLog
			//
			this.txtChangeLog.Anchor = (System.Windows.Forms.AnchorStyles) (((System.Windows.Forms.AnchorStyles.Top | System.Windows.Forms.AnchorStyles.Bottom) 
				| System.Windows.Forms.AnchorStyles.Left) 
				| System.Windows.Forms.AnchorStyles.Right);
			this.txtChangeLog.BackColor = System.Drawing.SystemColors.Control;
			this.txtChangeLog.BorderStyle = System.Windows.Forms.BorderStyle.None;
			this.txtChangeLog.Cursor = System.Windows.Forms.Cursors.Default;
			this.txtChangeLog.Font = new System.Drawing.Font("Microsoft Sans Serif", (float) (9.0F), System.Drawing.FontStyle.Regular, System.Drawing.GraphicsUnit.Point, System.Convert.ToByte(0));
			this.txtChangeLog.ForeColor = System.Drawing.SystemColors.ControlText;
			this.txtChangeLog.Location = new System.Drawing.Point(24, 224);
			this.txtChangeLog.Multiline = true;
			this.txtChangeLog.Name = "txtChangeLog";
			this.txtChangeLog.ReadOnly = true;
			this.txtChangeLog.ScrollBars = System.Windows.Forms.ScrollBars.Vertical;
			this.txtChangeLog.Size = new System.Drawing.Size(760, 192);
			this.txtChangeLog.TabIndex = 7;
			this.txtChangeLog.TabStop = false;
			//
			//lblTitle
			//
			this.lblTitle.AutoSize = true;
			this.lblTitle.Font = new System.Drawing.Font("Microsoft Sans Serif", (float) (14.0F), System.Drawing.FontStyle.Bold, System.Drawing.GraphicsUnit.Point, System.Convert.ToByte(0));
			this.lblTitle.ForeColor = System.Drawing.SystemColors.ControlText;
			this.lblTitle.Location = new System.Drawing.Point(16, 16);
			this.lblTitle.Name = "lblTitle";
			this.lblTitle.Size = new System.Drawing.Size(122, 27);
			this.lblTitle.TabIndex = 0;
			this.lblTitle.Text = "mRemoteNG";
			this.lblTitle.UseCompatibleTextRendering = true;
			//
			//lblVersion
			//
			this.lblVersion.AutoSize = true;
			this.lblVersion.Font = new System.Drawing.Font("Microsoft Sans Serif", (float) (11.0F), System.Drawing.FontStyle.Regular, System.Drawing.GraphicsUnit.Point, System.Convert.ToByte(0));
			this.lblVersion.ForeColor = System.Drawing.SystemColors.ControlText;
			this.lblVersion.Location = new System.Drawing.Point(16, 56);
			this.lblVersion.Name = "lblVersion";
			this.lblVersion.Size = new System.Drawing.Size(57, 22);
			this.lblVersion.TabIndex = 1;
			this.lblVersion.Text = "Version";
			this.lblVersion.UseCompatibleTextRendering = true;
			//
			//lblChangeLog
			//
			this.lblChangeLog.AutoSize = true;
			this.lblChangeLog.Font = new System.Drawing.Font("Microsoft Sans Serif", (float) (11.0F), System.Drawing.FontStyle.Regular, System.Drawing.GraphicsUnit.Point, System.Convert.ToByte(0));
			this.lblChangeLog.ForeColor = System.Drawing.SystemColors.ControlText;
			this.lblChangeLog.Location = new System.Drawing.Point(16, 199);
			this.lblChangeLog.Name = "lblChangeLog";
			this.lblChangeLog.Size = new System.Drawing.Size(92, 22);
			this.lblChangeLog.TabIndex = 6;
			this.lblChangeLog.Text = "Change Log:";
			this.lblChangeLog.UseCompatibleTextRendering = true;
			//
			//lblLicense
			//
			this.lblLicense.AutoSize = true;
			this.lblLicense.Font = new System.Drawing.Font("Microsoft Sans Serif", (float) (11.0F), System.Drawing.FontStyle.Regular, System.Drawing.GraphicsUnit.Point, System.Convert.ToByte(0));
			this.lblLicense.ForeColor = System.Drawing.SystemColors.ControlText;
			this.lblLicense.Location = new System.Drawing.Point(16, 96);
			this.lblLicense.Name = "lblLicense";
			this.lblLicense.Size = new System.Drawing.Size(58, 22);
			this.lblLicense.TabIndex = 5;
			this.lblLicense.Text = "License";
			this.lblLicense.UseCompatibleTextRendering = true;
			//
			//lblCopyright
			//
			this.lblCopyright.AutoSize = true;
			this.lblCopyright.Font = new System.Drawing.Font("Microsoft Sans Serif", (float) (11.0F), System.Drawing.FontStyle.Regular, System.Drawing.GraphicsUnit.Point, System.Convert.ToByte(0));
			this.lblCopyright.ForeColor = System.Drawing.SystemColors.ControlText;
			this.lblCopyright.Location = new System.Drawing.Point(16, 76);
			this.lblCopyright.Name = "lblCopyright";
			this.lblCopyright.Size = new System.Drawing.Size(70, 22);
			this.lblCopyright.TabIndex = 2;
			this.lblCopyright.Text = "Copyright";
			this.lblCopyright.UseCompatibleTextRendering = true;
			//
			//About
			//
			this.BackColor = System.Drawing.SystemColors.Control;
			this.ClientSize = new System.Drawing.Size(784, 564);
			this.Controls.Add(this.pnlTop);
			this.Controls.Add(this.pnlBottom);
			this.Font = new System.Drawing.Font("Microsoft Sans Serif", (float) (8.25F), System.Drawing.FontStyle.Regular, System.Drawing.GraphicsUnit.Point, System.Convert.ToByte(0));
			this.ForeColor = System.Drawing.SystemColors.ControlText;
			this.Icon = My.Resources.mRemote_Icon;
			this.MaximumSize = new System.Drawing.Size(20000, 10000);
			this.Name = "About";
			this.TabText = "About";
			this.Text = "About";
			this.pnlTop.ResumeLayout(false);
			this.pnlTop.PerformLayout();
			((System.ComponentModel.ISupportInitialize) this.pbLogo).EndInit();
			this.pnlBottom.ResumeLayout(false);
			this.pnlBottom.PerformLayout();
			this.ResumeLayout(false);
					
		}
        #endregion
				
        #region Public Methods
		public About(DockContent Panel)
		{
			this.WindowType = Type.About;
			this.DockPnl = Panel;
			this.InitializeComponent();
			App.Runtime.FontOverride(this);
		}
        #endregion
				
        #region Private Methods
		private void ApplyLanguage()
		{
			lblLicense.Text = My.Language.strLabelReleasedUnderGPL;
			lblChangeLog.Text = My.Language.strLabelChangeLog;
			TabText = My.Language.strAbout;
			Text = My.Language.strAbout;
		}
				
		private void ApplyEditions()
		{
            #if PORTABLE
			lblEdition.Text = My.Language.strLabelPortableEdition;
			lblEdition.Visible = true;
            #endif
		}
				
		private void FillLinkLabel(LinkLabel llbl, string Text, string URL)
		{
			llbl.Links.Clear();
					
			int Open = Text.IndexOf("[");
			int Close = 0;
			while (Open != -1)
			{
				Text = Text.Remove(Open, 1);
				Close = Text.IndexOf("]", Open);
				if (Close == -1)
				{
					break;
				}
				Text = Text.Remove(Close, 1);
				llbl.Links.Add(Open, Close - Open, URL);
				Open = Text.IndexOf("[", Open);
			}
					
			llbl.Text = Text;
		}
        #endregion
				
        #region Form Stuff
		private void About_Load(object sender, System.EventArgs e)
		{
			ApplyLanguage();
			ApplyEditions();
					
			try
			{
				lblCopyright.Text = (new Microsoft.VisualBasic.ApplicationServices.WindowsFormsApplicationBase()).Info.Copyright;
						
				this.lblVersion.Text = "Version " + (new Microsoft.VisualBasic.ApplicationServices.WindowsFormsApplicationBase()).Info.Version.ToString();
						
				FillLinkLabel(llblFAMFAMFAM, My.Language.strFAMFAMFAMAttribution, My.Language.strFAMFAMFAMAttributionURL);
				FillLinkLabel(llblMagicLibrary, My.Language.strMagicLibraryAttribution, My.Language.strMagicLibraryAttributionURL);
				FillLinkLabel(llblWeifenLuo, My.Language.strWeifenLuoAttribution, My.Language.strWeifenLuoAttributionURL);
						
				if (File.Exists((new Microsoft.VisualBasic.ApplicationServices.WindowsFormsApplicationBase()).Info.DirectoryPath + "\\CHANGELOG.TXT"))
				{
					StreamReader sR = new StreamReader((new Microsoft.VisualBasic.ApplicationServices.WindowsFormsApplicationBase()).Info.DirectoryPath + "\\CHANGELOG.TXT");
					this.txtChangeLog.Text = sR.ReadToEnd();
					sR.Close();
				}
			}
			catch (Exception ex)
			{
				Runtime.MessageCollector.AddMessage(Messages.MessageClass.ErrorMsg, "Loading About failed" + Environment.NewLine + ex.Message, true);
			}
		}
				
		private void llblFAMFAMFAM_LinkClicked(System.Object sender, System.Windows.Forms.LinkLabelLinkClickedEventArgs e)
		{
			App.Runtime.GoToURL(My.Language.strFAMFAMFAMAttributionURL);
		}
				
		private void llblMagicLibrary_LinkClicked(System.Object sender, System.Windows.Forms.LinkLabelLinkClickedEventArgs e)
		{
			App.Runtime.GoToURL(My.Language.strMagicLibraryAttributionURL);
		}
				
		private void llblWeifenLuo_LinkClicked(System.Object sender, System.Windows.Forms.LinkLabelLinkClickedEventArgs e)
		{
			App.Runtime.GoToURL(My.Language.strWeifenLuoAttributionURL);
		}
        #endregion
	}
}
